cmd/kubeadm/app/constants/constants.go

```diff
@@ -285,6 +285,16 @@ const (
 
 	// CoreDNSVersion is the version of CoreDNS to be deployed if it is used
 	CoreDNSVersion = "1.1.3"
+
+	// MasterConfigurationKind is the string kind value for the MasterConfiguration struct
+	MasterConfigurationKind = "MasterConfiguration"
+
+	// NodeConfigurationKind is the string kind value for the MasterConfiguration struct
+	NodeConfigurationKind = "NodeConfiguration"
+
+	// YAMLDocumentSeparator is the separator for YAML documents
+	// TODO: Find a better place for this constant
+	YAMLDocumentSeparator = "---\n"
```

Comment:
@luxas a couple of notes:

1) 
```
key: value---
```
i think the above would trigger the separator, where `---` could be intended as being part of `value`.

could be the safer way to do this.
`YAMLDocumentSeparator = "\n---\n"`

2)
`\n` would only handle the Unix (POSIX), LF case.
does the file loading instrumentation in k8s convert e.g CRLF (`\r\n`) line endings to LF `\n` automatically?
things like `ioutil.ReadFile()` do not, which means that if the user created his config file with CRLF endings it would look like this:

`\r\n---\r\n` == `13 10 45 45 45 13 10`

OSX sanely moved to the LF a while ago, while some editors on Windows are still CRLF by default.

Reply:
^ so it seems the reader comes from api machinery, so 1) is handled by using `\n---`. not ideal but it's better than `---\n`.
2) is not handled it seems.

the separator constant should be exported from here IMO:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/util/yaml/decoder.go#L139

Reply:
Practically all text editors on Windows (apart from Notepad) support Unix style line endings and UTF-8 these days, so I don't think that this is much of a problem here.

Reply:
that's true, but the parser has to be CRLF ready. otherwise - breakage.

Reply:
This is only used for serialization, not deserialization. So I'm not sure these comments apply

Reply:
yeah, this is fine for now.
i'm going to test the api machinery code in terms of CRLF and things like that, soon.

Reply:
great :+1: